src/ucs/sys/topo/base/topo.c

```diff
@@ -306,6 +372,65 @@ ucs_topo_get_distance_sysfs(ucs_sys_device_t device1,
     return UCS_OK;
 }
 
+ucs_numa_distance_t
+ucs_topo_numa_distance(ucs_numa_node_t source, ucs_numa_node_t dest)
+{
+    ucs_numa_distance_t distance;
+
+    ucs_spin_lock(&ucs_topo_global_ctx.lock);
+    distance = ucs_numa_distance(source, dest);
+    ucs_spin_unlock(&ucs_topo_global_ctx.lock);
+
+    return distance;
+}
+
+static ucs_status_t
+ucs_topo_get_memory_distance_sysfs(ucs_sys_device_t device,
+                                   ucs_sys_dev_distance_t *distance)
+{
+    double total_distance = 0;
+    ucs_sys_cpuset_t thread_cpuset;
+    int cpu, num_cpus;
+    char path[PATH_MAX];
+    ucs_numa_node_t dev_node;
+    ucs_status_t status;
+    int ret;
+
+    /* If the device is unknown, we assume min distance */
+    if (device == UCS_SYS_DEVICE_ID_UNKNOWN) {
+        return ucs_topo_get_memory_distance_default(device, distance);
+    }
+
+    ret = ucs_sys_pthread_getaffinity(&thread_cpuset);
+    if (ret) {
+        ucs_error("sched_getaffinity() failed: %m");
```

Comment:
ucs_sys_pthread_getaffinity should return ucs_status_t, and print the error inside of it.
because it's a UCS wrapper it should return UCS status and print the error itself

Reply:
I agree. I followed existing wrappers in ucs, `ucs_sys_setaffinity`, `ucs_sys_getaffinity`, they also return the return status of the subroutine...

Reply:
@yosefe if so, do we want to change the signature of existing `ucs_sys_{g,s}etaffinity` as well (separately) ?

Reply:
@yosefe  ?

Reply:
> @yosefe if so, do we want to change the signature of existing `ucs_sys_{g,s}etaffinity` as well (separately) ?

Yes, good idea. But in this PR pls update only the newly added functions.